fury/ui/helpers.py

```diff
@@ -23,6 +22,81 @@ def clip_overflow(textblock, width, side='right'):
     -------
     clipped text : str
         Clipped version of the text.
+    """
+    original_str = textblock.message
+    prev_bg = textblock.have_bg
+
+    _, mid_ptr = check_overflow(textblock, width, '...', side)
+
+    if mid_ptr == 0:
+        return original_str
+
+    textblock.have_bg = prev_bg
+    return textblock.message
+
+
+def wrap_overflow(textblock, wrap_width, side='right'):
+    """Wraps overflowing text of TextBlock2D with respect to width.
+
+    Parameters
+    ----------
+    textblock : TextBlock2D
+        The textblock object whose text needs to be wrapped.
+    wrap_width : int
+        Required width of the wrapped text.
+    side : str, optional
+        Clips the overflowing text according to side.
+        It takes values "left" or "right".
+
+    Returns
+    -------
+    wrapped text : str
+        Wrapped version of the text.
+    """
+    original_str = textblock.message
+    str_copy = textblock.message
+    prev_bg = textblock.have_bg
+    wrap_idx = []
+
+    _, mid_ptr = check_overflow(textblock, wrap_width, '', side)
+
+    if mid_ptr == 0:
+        return original_str
+
+    wrap_idx.append(mid_ptr)
+
+    while mid_ptr != 0:
+        str_copy = str_copy[mid_ptr:]
+        textblock.message = str_copy
+        _, mid_ptr = check_overflow(textblock, wrap_width, '', side)
+        if mid_ptr != 0:
+            wrap_idx.append(wrap_idx[-1]+mid_ptr+1)
+
+    for idx in wrap_idx:
+        original_str = original_str[:idx] + '\n' + original_str[idx:]
+
+    textblock.message = original_str
+    textblock.have_bg = prev_bg
+    return textblock.message
+
+
+def check_overflow(textblock, width, overflow_postfix='',
+                   side='right'):
+    """Checks if the text is overflowing.
+
+    Parameters
+    ----------
+    textblock : TextBlock2D
+        The textblock object whose text is to be checked.
+    width: int
+        Required width of the text.
+    overflow_postfix: str, optional
+        Postfix to be added to the text if it is overflowing.
+
+    Returns
+    -------
+    tuple(bool, int)
+
```

Comment:
You need to refactor the signature of this method as you never return `True` for this method.

Reply:
The method has been updated, there wasn't any purpose for the bool, now it returns just the overflow idx.